elements/pl-integer-input/pl-integer-input.py

```diff
@@ -37,9 +37,22 @@ def prepare(element_html, data):
         if name in data['correct_answers']:
             raise Exception('duplicate correct_answers variable name: %s' % name)
         # Test conversion, but leave as string so proper value is shown on answer panel
-        if pl.string_to_integer(correct_answer, base) is None:
+        a_tru_parsed = pl.string_to_integer(correct_answer, base)
+        if a_tru_parsed is None:
             raise Exception('correct answer is not a valid input: %s' % name)
+        if a_tru_parsed > 2 ** 53 - 1 or a_tru_parsed < -((2 ** 53) - 1):
+            raise Exception('answer must be between (2**53)-1 and -((2**53)-1)')
         data['correct_answers'][name] = correct_answer
+        return
+
+    a_tru = pl.from_json(data['correct_answers'].get(name, None))
+    if a_tru is not None:
+        if isinstance(a_tru, str):
+            a_tru = pl.string_to_integer(a_tru, base)
+        if a_tru is None:
+            raise Exception('correct answer is not a valid input: %s' % name)
+        if a_tru > 2 ** 53 - 1 or a_tru < -((2 ** 53) - 1):
+            raise Exception('answer must be between (2**53)-1 and -((2**53)-1)')
```

Comment:
Same comments as the ones @mwest1066 listed above for the other check. Also, I believe the other test above could be replaced with this one only, since you set `data["correct_answers"][name]` in line 45 anyway, so this would be checked for both cases anyway.

Reply:
line 45 wouldn't run if there's no html input wouldn't this need to be checked in case of the python input for example being a string or something?

Reply:
oh but if add one of the previous suggestions (casting directly to an int with a try block) I don't need to check for none because if it's not an int it'll throw an error

Reply:
Note that none is still a valid value, in that case we assume the question has a custom grader. What's not valid is a correct value that is not an integer in a valid range, or can't be converted to one.

Reply:
oh so I shouldn't raise an exception if it's None? only when out of bounds or can't be converted

Reply:
Yes, though note you already do a check for None in line 49.